src/shared/configUtils.ts

```diff
@@ -35,32 +61,36 @@ export class ConfigUtils {
     return this.#globalConfig;
   }
 
-  public static async getOrCreateIdentityToken(): Promise<string> {
-    let token = await this.getIdentityToken();
-    if (!token) {
-      token = CryptoUtils.generateIdentityToken();
-      await this.writeIdentityToken(token);
+  public static async getOrCreateIdentityToken(username: string, tokenService: IdentityTokenService): Promise<string> {
+    let identityData = await this.getIdentityData();
+    if (!identityData) {
+      const token = CryptoUtils.generateIdentityToken();
+      const entityId = await tokenService.saveTokenToServer(token);
+      identityData = {
+        identityToken: token,
+        usernameToServerEntityIdMap: {},
+      };
+      identityData.usernameToServerEntityIdMap[username] = entityId;
+      await this.writeIdentityData(identityData);
+      return token;
+    } else {
+      const entityId = await tokenService.saveTokenToServer(identityData.identityToken);
+      identityData.usernameToServerEntityIdMap[username] = entityId;
+      await this.writeIdentityData(identityData);
+      return identityData.identityToken;
     }
-    return token;
-  }
-
-  public static async getIdentityToken(): Promise<string | undefined> {
-    const config = await ConfigAggregator.create({ customConfigMeta: configMeta });
-    // Need to reload to make sure the values read are decrypted
-    await config.reload();
-    const identityToken = config.getPropertyValue(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_TOKEN);
-
-    return identityToken as string;
   }
 
-  public static async writeIdentityToken(token: string): Promise<void> {
+  public static async writeIdentityData(identityData: LocalWebServerIdentityData): Promise<void> {
     const config = await this.getConfig();
-    config.set(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_TOKEN, token);
+    // TODO: JSON needs to be stringified in order for config.write to encrypt. When config.write()
+    //       can encrypt JSON data to write it into config we shall remove stringify().
+    config.set(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_DATA, JSON.stringify(identityData));
     await config.write();
   }
 
   public static async getCertData(): Promise<SSLCertificateData | undefined> {
-    const config = await this.getGlobalConfig();
+    const config = await this.getConfig();
```

Comment:
Nit: I should have done this in my previous PR where I introduced `getGlobalConfig`.... but I think it would be a nice touch to cleanup this file and rename `#config` to `#localConfig` and `getConfig` to `getLocalConfig` to better align with the counterpart `getGlobalConfig`